build.gradle

```diff
@@ -1,93 +1,284 @@
-    def proc = cmd.execute()
-    proc.waitFor()
-    if(!ignoreError && proc.exitValue() != 0) {
-        println("Command '${cmd.join(" ")}' failed with exitCode ${proc.exitValue()}")
+task compile(type: Copy, dependsOn: goBuild) {
+    destinationDir = file('./bin')
+    for (p in platforms) {
+        from("./build/${p.goOs}-${p.goArch}") {
+            into("${p.owOs}/${p.goArch}")
+        }
+    }
+
+    /* Guarantee that the ./bin/wsk binary is executable */
+    from("./build/${rootProject.localPlatform.goOs}-${rootProject.localPlatform.goArch}") {
+        into('.')
     }
 }
 
-def getOsName() {
-    return System.properties['os.name']
+/*
+    Set up the goTest task to print useful information to Stdout:
+ */
+goTest.generateTestReport = { testClassResults ->
+    testClassResults*.results.flatten()*.failures.flatten().each() { failure ->
+        logger.quiet( '-' * 40 )
+        logger.quiet(failure.message)
+        logger.quiet( '-' * 40 )
+    }
 }
 
-def getOsArch() {
-    return System.properties['os.arch']
+/*
+    For each platform, create an individual archive in a platform appropriate
+    format (tarball for Linux, zipfile for Mac & Windows).
+ */
+task individualArchives(
+    dependsOn: rootProject.platforms.collect() { p ->
+        task("release${p.goOs.capitalize()}${p.goArch.capitalize()}",
+            type: (p.goOs == 'linux') ? Tar : Zip, dependsOn: compile) {
+                if (p.goOs == 'linux') { compression = Compression.GZIP }
+                destinationDir = file('./release')
+                baseName = "${p.zipFileName}-latest-${p.owOs}-${p.goArch}"
+                from "./build/${p.goOs}-${p.goArch}/"
+            }
+    })
+
+/*
+    Create a 'content.json' file representing all that was
+    compiled and its appropriate directory in the Tarball that will be created
+    for deployment to local Nginx instances.
+ */
+
+task index() {
+    def content = [:]
+    for (p in platforms) {
+        def pathObject = [ "path" : "${p.archiveDirName}/${p.archiveFileName}" ]
+        content.get(p.owOs,[:])[p.goArch] = pathObject
+        // TODO: Default architecture should be configurable as a property
+        if (p.goArch == 'amd64') {
+            content.get(p.owOs,[:])['default'] = pathObject
+        }
+    }
+
+    doLast {
+        mkdir('./build')
+        file('./build/content.json').text = groovy.json.JsonOutput.toJson(["cli": content])
+    }
 }
 
-def mapOsNameToGoName(String osname) {
-    String osname_l = osname.toLowerCase()
-    if (osname_l.contains("nux") || osname.contains("nix")) return "linux"
-    if (osname_l.contains("mac")) return "mac"
-    if (osname_l.contains("windows")) return "windows"
-    return osname_l
+task release(type: Tar, dependsOn: [individualArchives, index]) {
+    compression = Compression.GZIP
+    destinationDir = file('./release')
+    baseName = "${OpenWhiskPlatform.zipFileName}-latest-all"
+    from('./release/content.json') { into('.') }
```

Comment:
`content.json` is now lives in the build folder.

Reply:
Fixed.